src/storage/datacore/restapi/mode/alertscount.pm

```diff
@@ -0,0 +1,171 @@
+            perfdatas       => [ { template => '%d', min => 0 } ]
+        }
+    }, {
+        label  => 'info',
+        nlabel => 'datacore.alerts.info.count',
+        set    => {
+            key_values      => [ { name => 'info' } ],
+            output_template => 'number of info alerts : %s',
+            perfdatas       => [ { template => '%d', min => 0 } ]
+        }
+    }, {
+        label  => 'trace',
+        nlabel => 'datacore.alerts.trace.count',
+        set    => {
+            key_values      => [ { name => 'trace' } ],
+            output_template => 'number of trace alerts : %s',
+            perfdatas       => [ { template => '%d', min => 0 } ]
+        }
+    },
+    ];
+
+}
+
+sub manage_selection {
+    my ($self, %options) = @_;
+
+    my $data = $options{custom}->request_api(
+        url_path => '/RestService/rest.svc/1.0/alerts'
+    );
+    my $alerts_count = $self->order_alerts($data);
+    $self->{alerts}->{trace} = $alerts_count->{$alerts_level{trace}}->{count};
+    $self->{alerts}->{info} = $alerts_count->{$alerts_level{info}}->{count};
+    $self->{alerts}->{warning} = $alerts_count->{$alerts_level{warning}}->{count};
+    $self->{alerts}->{error} = $alerts_count->{$alerts_level{error}}->{count};
+
+}
+
+# take a decoded json reference and send back a hash with loglevel as key and a hash with number of alert
+# and the list of alerts text as value.
+sub order_alerts {
+    my ($self, $alerts) = @_;
+    my %alerts_count = (
+        0 => { count => 0, list => [] },
+        1 => { count => 0, list => [] },
+        2 => { count => 0, list => [] },
+        3 => { count => 0, list => [] });
+
+    for my $alert (@$alerts) {
+
+        # spec require to filter on time of the log.
+        $alert->{TimeStamp} =~ /\/Date\((\d+)\)\/$/;
+        my $alert_date = $1;
+        # filter on age of the alert with a user defined max age
+        next if (defined($self->{option_results}->{max_alert_age})
+            and $alert_date < (time - $self->{option_results}->{max_alert_age}) * 1000);
+        # filter on the machine issuing the alert with a user defined regex
+
+        if (!centreon::plugins::misc::empty($self->{option_results}->{filter_server})
+            and $alert->{MachineName} !~ $self->{option_results}->{filter_server}) {
+            $self->{output}->output_add(long_msg => "excluding alert from machine $alert->{MachineName}\n", debug => 1);
+            next;
+        }
+
+        $alerts_count{$alert->{Level}}->{count}++;
+        # we don't want to clog the long output, so we keep only the few first logs.
+        # we use a array instead of directly adding to long_output because we need to sort errors
+        if (scalar(@{$alerts_count{$alert->{Level}}->{list}}) < 50) {
+            push(@{$alerts_count{$alert->{Level}}->{list}}, $alert->{MessageText})
+        }
+    }
+
+    $self->{output}->output_add(long_msg => "error : " . join("\n", @{$alerts_count{$alerts_level{error}}->{list}}));
```

Comment:
Following my previous comment: shouldn't we cut in that place? Once the quota is reached, no more output_add?